nixpkgs/nixpkgs.bzl

```diff
@@ -175,14 +178,169 @@ def nixpkgs_package(*args, **kwargs):
     if "repositories" in kwargs:
         inversed_repositories = {value: key for (key, value) in kwargs["repositories"].items()}
         kwargs.pop("repositories")
-        _nixpkgs_package(
+        f(
             repositories = inversed_repositories,
             *args,
             **kwargs
         )
     else:
         _nixpkgs_package(*args, **kwargs)
 
+def _nixpkgs_packages_instantiate_impl(repository_ctx):
+    repository_ctx.file("BUILD", "exports_files([\"nix_attrs.nix\"])")
+
+    nix_instantiate_path = _executable_path(
+        repository_ctx,
+        "nix-instantiate",
+        extra_msg = "See: https://nixos.org/nix/",
+    )
+
+    packages_record_inside = ";".join(
+        [
+            "\"{name}\" = nixpkgs.{attrPath}".format(name = name, attrPath = attrPath)
+            for (name, attrPath) in repository_ctx.attr.packages.items()
+        ]
+    )
+    packages_record = "nixpkgs: { " + packages_record_inside + "; }"
+
+    repository_ctx.file("packages_attributes_mappings.nix", packages_record)
```

Comment:
is this path unique enough that two nixpkgs_packages can be defined in the same project?

Reply:
Yes, that path is namespaced by the value of the `name` attribute of the rule which has to be unique